### PR TITLE
[doc] Model card example, template and instructions

### PR DESCRIPTION
**Pull request by stas00**
As discussed here: https://discuss.huggingface.co/t/tips-for-debugging-model-cards/814 this is a proposal which may make it easier to create model_cards.

This PR splits the all-in-one approach which is difficult to use into 3:

1.  a template with just fields and no values
2.  a complete demo template that shows a very perfect entry with multiple entries where array is expected/supported
3.  a document that explains each field’s format and variations.

Surely, what I made could use more work, but this is a start.

### CODE REVIEW

File: README.md
Comment (sshleifer) [inline README.md]:
```suggestion
One of the valid license identifiers. e.g.:
```

File: README.md
Comment (sshleifer) [inline README.md:69]: zero or more

File: README.md
Comment (sshleifer) [inline README.md:34]: zero or more

**Review by sshleifer — APPROVED** [review]
Thanks for this. I think it moves the project forward. We will wait for @julien-c to merge.

Should be linked to from https://github.com/huggingface/transformers/blob/master/templates/adding_a_new_model/README.md
after merging.

File: README.md
Comment (stas00) [reply README.md:69]: you commented and then resolved it, perhaps by mistake? so change to zero or not?

File: demo.README.md
Comment (julien-c) [inline demo.README.md:9]: see https://huggingface.co/docs#what-metadata-can-i-add-to-my-model-card

> License identifiers are those standardized by GitHub here.

Comment (stas00) [reply demo.README.md:9]: Right, see I'm still wrong. 

"License identifiers are those standardized by GitHub here." is very vague - do we want the keyword "apache-2.0" or the full string "Apache license 2.0"?

Have a look at the current cards:

```
$ grep -hr license: model_cards | sort | uniq
license: ""
license:  Apache-2.0
license: Apache 2.0
license: Apache-2.0
license: "apache-2.0"
license: apache-2.0
license: cc-by-nc-4.0
license: MIT
license: "MIT"
license: mit
license: "mit"
license: mit
```

Different people interpreted it differently. And not a single one has:  "Apache license 2.0"

**Comment by stas00** [comment]
Also, to clarify, I'm totally not attached to this particular way I suggested. 

As long as there is enough of **precise** instructions and some correct examples, that should work.

**Comment by julien-c** [comment]
> Also, to clarify, I'm totally not attached to this particular way I suggested.
> 
> As long as there is enough of **precise** instructions and some correct examples, that should work.

Understood! And thanks for contributing. I'll think of how to make things clearer in the next few weeks

File: demo.README.md
Comment (julien-c) [reply demo.README.md:9]: yaml removes non-necessary quotes, and I guess we could auto-lowercase, so there's no that much variation there

Again, mid-term goal is to have built-in validation!

Comment (julien-c) [reply demo.README.md:9]: Oh yes and by identifiers I meant the keywords in the keyword column: https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/licensing-a-repository#searching-github-by-license-type

Comment (stas00) [reply demo.README.md:9]: If I may share my experience - The problem is not so much whether one could automate the normalization of tags. The problem is that it's very confusing to the writer of the card of what is expected, especially when there is such a wide variety of tags that do the same. Should I copy from this file, or that one. Hence my suggestion would be to fix all the cards to do it in an identical way - i.e. above there should be only 3 licenses - and not 9. 

Thank you.

**Comment by stas00** [comment]
For the demo file if I may make a suggestion, you could pick one of the most complex existing cards, one that has multiple languages, multiple datasets, etc., make it shine and use that as a demo page. Or perhaps a few showcases, each with distinct features. It's often much easier to copy and modify something than follow instructions starting with a blank page.

Please feel free to close this PR if you feel that a different approach would be a better choice.

**Comment by stas00** [comment]
one more issue - `thumbnail` - any restrictions on that?
e.g. I found this large image: https://github.com/pytorch/fairseq/blob/master/docs/fairseq_logo.png (direct link https://raw.githubusercontent.com/pytorch/fairseq/master/docs/fairseq_logo.png)
but I don't know if that would work or not - again I have to try and merge 

since I didn't have a thumbnail I found this guy https://github.com/pytorch/fairseq/blob/master/docs/fairseq_logo.png (direct link https://raw.githubusercontent.com/pytorch/fairseq/master/docs/fairseq_logo.png)

**Comment by julien-c** [comment]
no restriction but you need to use a direct link

**Comment by mapmeld** [comment]
@stas00 I was asked to look at this PR. Thanks for writing this up

- all fields are voluntary, so I wonder if instead of "Here is how to create a model_card" that section could be a little friendlier, such as, "We recommend sharing information about each model in a model_card.  Get started with our template card:"

- could the YAML metadata section of README.md start with a short \`\`\`code block\`\`\` showing sample YAML, and then your explanation of how language, license, and other fields are filled
```yaml
license: mit
language:
- es
- ja
``` 
- the metrics field isn't supported yet - should it work like the dataset tags?  I've asked HuggingFace about including more information, such as a reported score. Do you think this would be useful?

```yaml
metrics:
  - name: sample_metric
    score: 15
```

**Comment by stas00** [comment]
Thank you for the suggestions, @mapmeld!

>  all fields are voluntary, so I wonder if instead of "Here is how to create a model_card" that section could be a little friendlier, such as, "We recommend sharing information about each model in a model_card.  Get started with our template card:"

Done

>  could the YAML metadata section of README.md start with a short ```code block``` showing sample YAML, and then your explanation of how language, license, and other fields are filled

Done

>  the metrics field isn't supported yet - should it work like the dataset tags?  I've asked HuggingFace about including more information, such as a reported score. Do you think this would be useful?

I think it is useful information. For example, one may want to browse models that use a specific metric to test quality or acceptance of a certain metric.

We can instruct to fill it out this field, and it'll be there when it will become supported. It'd be easier to normalize the names, (which surely will vary unless we give a fixed list of metrics) than to ask to re-add them later to the ever growing list of cards.

p.s. please feel free to make direct suggestions to any wording you feel could be improved. The main intention of this PR was to compile/provide technical information, with little attention to specific wording, as i wasn't even sure this PR would be accepted.

File: README.md
Comment (julien-c) [inline README.md]:
```suggestion
- apache-2.0
- mit 
```

File: README.md
Comment (julien-c) [inline README.md]:
```suggestion
license: apache-2.0
```

File: README.md
Comment (julien-c) [inline README.md:7]:
```suggestion
The canonical documentation about model cards is at https://huggingface.co/docs

One of the easiest way to get started is by using our template card. Simply copy `template.README.md` to `model_cards/USERNAME/MODELNAME/README.md` and fill it out while studying [demo.README.md](./demo.README.md) for an example of a model `model_card` and referring to rest of this short document if you are not sure about some specific fields.
```

**Review by julien-c — COMMENTED** [review]
Thank you

**Comment by julien-c** [comment]
Thanks @stas00, I've also made some updates to the canonical doc page at https://huggingface.co/docs (internal repo) to make a few things clearer

Screenshot 2020-10-15 at 23 03 53

Feel free to update further updates, in particular to that doc page

**Comment by stas00** [comment]
Looking great, @julien-c! Thank you for adding the extra clarifications!

I will revisit it again when I add a new model_card

**Pull request merged**
